### ISSUE TITLE
Normalisation layer save standard deviation instead of variance

### ISSUE THREAD

**Issue by blubbrezn** [tensorflow/tensorflow#46266]
**System information**
- TensorFlow version (you are using): 2.2.0
- Are you willing to contribute it (Yes/No): No

**Describe the feature and the current behavior/state.**
The Normalisation layer in tensorflow/python/keras/layers/preprocessing/normalization.py seems to save the variance instead of the standard deviation. And calculates the std on runtime std = sqrt(var).
This seems a bit wastfull to me as instead of saving the std once.
It would be a very small change with not much impact, but easy to implement.

**Will this change the current api? How?**
When using adapt() to initialize it doesn't matter which metric is saved -> no change in api
When directly initializing it, it could still take the variance (or std as an option) and save std in the background

**Any Other info.**
Could be changed at the next big change e.g: together with the behaviour I stated in [keras #14356](https://github.com/keras-team/keras/issues/14356#issue-781559372)

**Comment by fchollet** [comment]
We should not change the weight format (in particular for backwards compatibility and because we want the weights of multiple normalizable layers to be easily mergeable), but what we could do is compute the `sqrt` in the constructor of the layer (or upon weight loading) and store it on the layer.

**Comment by tilakrayal** [comment]
@blubbrezn,
Batch normalization applies a transformation that maintains the mean output close to 0 and the output standard deviation close to 1.

During training (i.e. when using fit() or when calling the layer/model with the argument training=True), **the layer normalizes its output using the mean and standard deviation of the current batch of inputs.**

During inference (i.e. when using evaluate() or predict() or when calling the layer/model with the argument training=False (which is the default), **the layer normalizes its output using a moving average of the mean and standard deviation of the batches it has seen during training**.  Kindly find the official doc for the [reference](https://keras.io/api/layers/normalization_layers/batch_normalization/). Thank you!
